app/components/primer/beta/auto_complete.rb

```diff
@@ -85,6 +93,35 @@ def initialize(label_text:, src:, list_id:, input_id:, is_label_visible: true, i
       # add `results` without needing to explicitly call them in the view
       def before_render
         results(classes: "") unless results
+        input unless input
+      end
+    end
+
+    # This component is part of `Primer::Beta::AutoCompleteComponent` and should not be
+    # used as a standalone component.
+    class Input < Primer::Component
+      # TODO: Do we fix this to always use `text`?
+      DEFAULT_TYPE = :text
+      TYPE_OPTIONS = [DEFAULT_TYPE, :search].freeze
+
+      # TODO: Decide if we want to allow aria-labels here?
+
+      # @param type [Symbol] <%= one_of(Primer::Beta::AutoComplete::Input::TYPE_OPTIONS) %>
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      def initialize(type: DEFAULT_TYPE, **system_arguments)
+        @system_arguments = deny_tag_argument(**system_arguments)
+        @system_arguments = deny_single_argument(:autofocus, "autofocus is not allowed for accessibility reasons.", **system_arguments)
```

Comment:
Should we be more explicit about `accessibility reasons`? Maybe point to a resource if one exists?